tests/imfile-statefile-delete.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+# added 2019-02-28
+# This is part of the rsyslog testbench, licensed under ASL 2.0
+. ${srcdir:=.}/diag.sh init
+export TESTMESSAGES=1000
+export TESTMESSAGESFULL=999 
+export RETRIES=50
+generate_conf
+add_conf '
+global(workDirectory="'${RSYSLOG_DYNNAME}'.spool")
+module(load="../plugins/imfile/.libs/imfile" mode="inotify" PollingInterval="1")
+input(type="imfile" tag="file:" file="./'$RSYSLOG_DYNNAME'.input")
+
+template(name="outfmt" type="string" string="%msg:F,58:2%\n")
+if $msg contains "msgnum:" then
+	action(type="omfile" file=`echo $RSYSLOG_OUT_LOG` template="outfmt")
+'
+./inputfilegen -m $TESTMESSAGES > $RSYSLOG_DYNNAME.input
+inode=$(get_inode "$RSYSLOG_DYNNAME.input")
+startup
+wait_file_lines $RSYSLOG_OUT_LOG $TESTMESSAGES $RETRIES
+rm $RSYSLOG_DYNNAME.input
+# sleep a little to give rsyslog a chance to notice the deleted file
```

Comment:
timing via sleep is racy and results in flakes. Do we have a chance to synchronize the wait in a different way?

Reply:
Hi @rgerhards, thanks for looking into this PR. I remove the 2s sleep from `imfile-statefile-delete.sh`. It now waits until the state file is deleted, and if it waits for longer than 30s the test fails.

  `imfile-state-file-no-delete.sh` needs to wait long enough for rsyslog to notice the file was deleted and decide not to delete the state file because `deleteStateOnFileDelete="off”`. I haven't been able to think of a better way of doing this yet, but if you are able to point me in the right direction I'd be happy to change that test too. I decided to use sleeps in the tests after looking at `imfile-wildcards-dirs-multi.sh` and `imfile-endregex.sh` that do a similar thing.

Reply:
it's indeed tricky, I agree. Sounds like a good solution. Let's see what CI now says. I am willing to merge and see if this test becomes racy in the long term. If so, we (I ;-)) need to think of a different way to handle it.